Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -130,6 +136,35 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
         AppEnvironment.current.environmentType.stripePublishableKey
       )
     }
+
+    // MARK: Validate Checkout Details On Submit
+
+    let stripeCardIdAndPaymentIntent = self.creditCardSelectedSignal.signal
+
+    let validateCheckout = Signal.combineLatest(checkoutId, stripeCardIdAndPaymentIntent)
+      .takeWhen(self.submitButtonTappedProperty.signal)
+      .switchMap { checkoutId, stripeCardIdAndPaymentIntent in
+        let (stripeCardId, paymentIntent) = stripeCardIdAndPaymentIntent
+
+        return AppEnvironment.current.apiService
+          .validateCheckout(
+            checkoutId: checkoutId,
+            paymentSourceId: stripeCardId,
+            paymentIntentClientSecret: paymentIntent
+          )
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .materialize()
+      }
+
+    self.validateCheckoutSuccess = stripeCardIdAndPaymentIntent
+      .takeWhen(validateCheckout.values().ignoreValues())
+      .map { stripeCardIdAndPaymentIntent in
```

Comment:
Nit: Can be slightly more compact like
`.map { stripeCardId, paymentIntent  in paymentIntent }`